pkgs/top-level/all-packages.nix

```diff
@@ -21191,7 +21191,10 @@ with pkgs;
 
   protobuf = protobuf3_19;
 
-  protobuf3_21 = callPackage ../development/libraries/protobuf/3.21.nix { };
+  # Versioning for protobuf changed scheme during 3.21.x releases
+  # See: https://developers.google.com/protocol-buffers/docs/news/2022-05-06
+  protobuf_21 = callPackage ../development/libraries/protobuf/21.nix { };
+  protobuf3_21 = protobuf_21;
```

Comment:
This is a tricky one.  I think maybe we don't need this alias?  But I don't feel strongly.

Reply:
```suggestion
```

Reply:
Upstream is currently publishing tags using the old and new model https://github.com/protocolbuffers/protobuf/tags.

In favor of removing: We haven't had a stable release with 3.21 in it yet, so not having it is just part of the pain of a rolling release.
In opposition to removing: Some out-of-tree people might already be using it, and having a hard eval failiure is a pretty awful experience. We should have an alias to reflect upstream behavior.

Reply:
> having a hard eval failiure is a pretty awful experience.

Normally we do not really care about versioned aliases though.

Reply:
depends on ecosystem and what we want to support

Reply:
Aliases only delay a hard failure introduced by the throw migration. At some point someone will have an eval failure as it stands.

Reply:
Going to leave the "alias" in all-packages, as this aligns with upstream cutting tags on both `3.21` and `21.x`